File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        std::unordered_map<char, int> count;
        int res = 0;
        int left = 0;
        for (int right = 0; right < s.size(); right++) {
            count[s[right]] = 1 + count[s[right]];

            int maxFrequency = 0;
            for (const auto& [key, value] : count) {
                maxFrequency = std::max(maxFrequency, value);
            }

            while ((right - left + 1) - maxFrequency > k) {
                count[s[left]] = count[s[left]] - 1;
                left++;
            }

            res = std::max(res, right - left + 1);
        }
        return res;

    }
};

File: Data Structures & Algorithms/max-consecutive-ones-ii/submission-0.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        l = 0
        k = 1
        res = 0
        n = len(nums)
        for r in range(n):
            if nums[r] == 0:
                k -= 1
            while k < 0:
                if nums[l] == 0:
                    k += 1
                l += 1
            res = max(res, r - l + 1)
        return res
